rct229/rulesets/ashrae9012019/section11/section11rule8.py

```diff
@@ -0,0 +1,248 @@
+            ]
+
+        class SWHBATRule(RuleDefinitionBase):
+            def __init__(self):
+                super(Section11Rule8.RMDRule.SWHBATRule, self).__init__(
+                    rmds_used=produce_ruleset_model_description(
+                        USER=False, BASELINE_0=True, PROPOSED=True
+                    ),
+                )
+
+            def is_applicable(self, context, data=None):
+                building_area_type_and_uses_p_b = context.PROPOSED
+                service_water_heating_uses_p = data["service_water_heating_uses_p"]
+
+                return all(
+                    [
+                        service_water_heating_uses_p[swh_uses_id_p] > 0.0
+                        for swh_uses_id_p in building_area_type_and_uses_p_b[
+                            "swh_bats_and_uses_p"
+                        ]
+                    ]
+                )
+
+            def get_calc_vals(self, context, data=None):
+                swh_bats_and_equip_dict_this_use = context.BASELINE_0
+                num_of_bldg_segment_b = data["num_of_bldg_segment_b"]
+                swh_bats_and_uses_b = data["swh_bats_and_uses_b"]
+
+                swh_bat_b = swh_bats_and_equip_dict_this_use["id"]
+
+                num_swh_systems_b = data["num_swh_systems_b"][swh_bat_b]
+                num_other_swh_bat_segment_b = data["num_other_swh_bat_segment_b"]
+
+                is_referenced_in_other_bats_b = True
+                if num_swh_systems_b == 1:
+                    is_referenced_in_other_bats_b = False
+                    swh_dist_id = swh_bats_and_uses_b[swh_bat_b].swh_distribution[0]
+                    for other_swh_bat in swh_bats_and_uses_b:
+                        if other_swh_bat != swh_bat_b:
+                            if (
+                                swh_dist_id
+                                in swh_bats_and_uses_b[swh_bat_b].swh_distribution
+                            ):
+                                is_referenced_in_other_bats_b = True
+
+                multiple_segments_with_bat_other_b = (
+                    True if num_other_swh_bat_segment_b > 1 else False
+                )
+
+                return {
+                    "swh_bat_b": swh_bat_b,
+                    "num_swh_systems_b": num_swh_systems_b,
+                    "num_of_bldg_segment_b": num_of_bldg_segment_b,
+                    "is_referenced_in_other_bats_b": is_referenced_in_other_bats_b,
+                    "multiple_segments_with_bat_other_b": multiple_segments_with_bat_other_b,
+                }
+
+            def manual_check_required(self, context, calc_vals=None, data=None):
+                swh_bat_b = calc_vals["swh_bat_b"]
+                multiple_segments_with_bat_other_b = calc_vals[
+                    "multiple_segments_with_bat_other_b"
+                ]
+
+                return swh_bat_b == "UNDETERMINED" or multiple_segments_with_bat_other_b
```

Comment:
`(swh_bat_b  == UNDETERMINED and num_of_bldg_segment_b > 1) or (swh_bat_b == OTHER and multiple_segments_with_bat_other_b > 1)`

Reply:
Good point. Addressed.